crates/wysiwyg/src/composer_model/delete_text.rs

```diff
@@ -224,18 +224,12 @@ where
             };
         }
         match self.state.dom.lookup_node_mut(&location.node_handle) {
-            // we should never be passed a container
-            DomNode::Container(_) => ComposerUpdate::keep(),
-            DomNode::LineBreak(_) => {
-                // for a linebreak, remove it if we started the operation from the whitespace
-                // char type, otherwise keep it
-                match start_type {
-                    CharType::Whitespace => self.delete_to_cursor(
-                        direction.increment(location.index_in_dom()),
-                    ),
-                    _ => ComposerUpdate::keep(),
-                }
-            }
+            DomNode::Container(_) | DomNode::LineBreak(_) => match start_type {
```

Comment:
The function to add line breaks is deprecated and fell out of use at the point of doing https://github.com/matrix-org/matrix-rich-text-editor/pull/472 I believe, so the Dom shouldn't contain these nodes. However, the `LineBreak` node is still being used to represent line breaks in the DOM immediately after parsing, before being post-processed into paragraphs. 

This is a bit problematic because although we can expect that the DOM should no longer contain `LineBreak`s, there is a lot of existing code that expects that it might which is effectively dead code and a maintenance burden.

Perhaps we can either
1. Remove the `LineBreak` node and parse `<br>` directly to `Container`. It will be the most effort but, after this PR, we have compatible tests to make this easier. There is also the question of whether this would put too much logic in the HTML parser.
2. Keep the `LineBreak` node but refactor it to something like `Placeholder(LineBreak)`, adding an invariant assertion that it should not exist in the Dom. We could remove any logic that handles this node type and replace it with an `unreachable!` error.

What do you think?

Reply:
Thanks for that. I'm ok with either option, was more trying to understand it's current use.

Reply:
https://github.com/matrix-org/matrix-rich-text-editor/issues/847